nautobot/core/views/mixins.py

```diff
@@ -536,6 +537,15 @@ def list(self, request, *args, **kwargs):
         """
         List the model instances.
         """
+        per_page = request.GET.get("per_page") or request.user.get_config(
+            "pagination.per_page", get_settings_or_config("PAGINATE_COUNT")
+        )
+        max_page_size = get_settings_or_config("MAX_PAGE_SIZE")
+        if int(per_page) > int(max_page_size):
+            messages.error(
+                request,
+                f"Pagination `per_page` has exceeded the maximum page size of {max_page_size}, triggering a return of {max_page_size} items.",
+            )
```

Comment:
Is this the best place to implement this or would it be better to put somewhere in views/renderers.py, like maybe `construct_table()`?